File: leetcode/01_99/82__Remove_Duplicates_from_Sorted_List_2.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode dummy = new ListNode(0);
        dummy.next = head;
        
        ListNode prev = dummy, cur = head;
        while(cur != null) {
            int count = 1;
            while(cur.next != null && cur.val == cur.next.val) {
                count++;
                cur = cur.next;
            }
            
            if (count == 1) {
                prev = cur;
            } else {
                prev.next = cur.next;
            }
            cur = cur.next;
        }
        return dummy.next;
    }
}

File: leetcode/01_99/82__Remove_Duplicates_from_Sorted_List_2.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */

typedef struct ListNode ListNode_t;

struct ListNode* deleteDuplicates(struct ListNode* head){
    ListNode_t dummy = {0, head};
    
    ListNode_t * prev = &dummy;
    ListNode_t * cur = head;
    while(cur != NULL) {
        int count = 1;
        while(cur->next != NULL && cur->val == cur->next->val) {
            count++;
            cur = cur->next;
        }
        if (count == 1) {
            prev = cur;
        } else {
            prev->next = cur->next;
        }
        cur = cur->next;
    }
    return dummy.next;
}

